lib/SensorFusion/src/SensorFusion.cpp

```diff
@@ -0,0 +1,112 @@
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  SensorFusion algorithm
+ * @author Juliane Kerpe <juliane.kerpe@web.de>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "SensorFusion.h"
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+/******************************************************************************
+ * Protected Methods
+ *****************************************************************************/
+
+/******************************************************************************
+ * Private Methods
+ *****************************************************************************/
+
+void SensorFusion::init(void)
+{
+    m_lkf.init();
+}
+
+void SensorFusion::estimateNewState(SensorData newSensorData)
+{
+    /* Estimate the current angle. */
+    int16_t estimatedAngle;
+    estimateAngle(estimatedAngle, newSensorData.angleOdometry, newSensorData.magnetometerValueX, newSensorData.magnetometerValueY);
+
+    /* Transform the acceleration values from the robot coordinate system into the world coordinate system */
+    int16_t accelerationInRobotCoordinateSystem[2] = {newSensorData.accelerationX, newSensorData.accelerationY};
+    int16_t accelerationInGlobalCoordinateSystem[2];
+    transfromLocalToGlobal(accelerationInGlobalCoordinateSystem, accelerationInRobotCoordinateSystem, estimatedAngle);
+
+    /* perform the Kalman Filter Prediction and Update Steps */
+    m_lkf.predictionStep();
+    m_lkf.updateStep(newSensorData.accelerationX, newSensorData.accelerationY, newSensorData.positionOdometryX, newSensorData.positionOdometryY);
+}
+
+void SensorFusion::transfromLocalToGlobal(int16_t* globalResult, const int16_t* localVectorToTransform, const int16_t& rotationAngle)
```

Comment:
This could also be declared as
```suggestion
void SensorFusion::transfromLocalToGlobal(int16_t globalResult[], const int16_t localVectorToTransform[], const int16_t& rotationAngle)
```

Technically, this is the same, but it makes more clear that an array is passed. Attention: Just like for the pointer notation, there is no size-safety! (Internally, there's still the pointer passed.)

Reply:
Leave it as a pointer since it fits the coding style of the rest of the Application (see e.g. RemoteControl/App.cpp/App_lineSensorChannelCallback)